### ISSUE TITLE
java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open(Ljava/lang/String;I)V

### ISSUE THREAD

**Issue by chsreedhar** [xerial/sqlite-jdbc#130]
Getting following error while connecting to database.
Tried with **sqlite-jdbc-3.8.11.1.jar**, **sqlite-jdbc-3.8.11.2.jar**, **sqlite-jdbc-3.8.6.jar**

```
java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open(Ljava/lang/String;I)V
    at org.sqlite.core.NativeDB._open(Native Method)
    at org.sqlite.core.DB.open(DB.java:161)
    at org.sqlite.core.CoreConnection.open(CoreConnection.java:145)
    at org.sqlite.core.CoreConnection.<init>(CoreConnection.java:66)
    at org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:21)
    at org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:23)
    at org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:45)
    at org.sqlite.JDBC.createConnection(JDBC.java:114)
    at org.sqlite.JDBC.connect(JDBC.java:88)
```

**Comment by gitblit** [comment]
Duplicate of #97.  Since SQLite is a hybrid of Java and native code, not having any architectural information about the target system makes this report inadequate.  At a minimum we need operating system, target architecture, and Java vendor.

**Comment by lex-divert** [comment]
The reported symptom looks like #97 , but that ticket's discussion ended with a workaround that does not fix the problem on ARMv7 platforms [1]. 

Specifically, the incorrect native driver is being selected, as described by @gnosygnu on the XOWA project's tracker: https://github.com/gnosygnu/xowa/issues/27#issuecomment-210243055 .

I've tried XOWA's build of sqlite-jdbc-3.8.11.2.jar and it does work for me. A quick investigation shows only one obvious difference:

File | Upstream md5 | XOWA md5
---- | --------------- | -----------
org/sqlite/native/ARM/libsqlitejdbc.so | 27bb60b6fe2196fec44c8e72369bc365 | __27bb60b6fe2196fec44c8e72369bc365__
org/sqlite/native/Linux/arm/libsqlitejdbc.so | c5d63c866ed6141dfac921366e7c2171 | c5d63c866ed6141dfac921366e7c2171
org/sqlite/native/Linux/armhf/libsqlitejdbc.so | 4c0b2ab37bfc41fde0c5e0af2596b7a7 | __27bb60b6fe2196fec44c8e72369bc365__

So it looks like the "Linux/armhf" driver was simply replaced with the "ARM" driver.

----
[1] In my case, if it matters, it's a Pandaboard running debian 7.9:
 • model name: ARMv7 Processor rev 10 (v7l)
 • Features: half thumb fastmult vfp edsp thumbee neon vfpv3 tbs vfpd32
• Hardware: Generic OMAP4 (Flattened Device Tree)
• uname -a: Linux debian 3.19.0-armv7-x3 #1 SMP Wed Feb 11 21:58:07 UTC 2015 armv7l GNU/Linux
• Java(TM) SE Runtime Environment (build 1.8.0_77-b03)

**Comment by lex-divert** [comment]
I guess this was mostly covered in #58 and its predecessors. Hopefully the eventual release of 3.13.0 will fix the confusion.

**Comment by yonisolo** [comment]
Yes this has been done because of the manner sqlite drivers get dynamicaly the good library from the cpu arch and the targeted OS. In the fact the ARM one would never be used and may be removed.

**Comment by xerial** [comment]
Duplicate of #188. Let me close this. This should be fixed in #193
